### ISSUE TITLE
Soundness hole: Arcu<T, P> implements Sync despite T not being Sync

### ISSUE THREAD

**Issue by adri326** [Skgland/Arcu#1]
Since `Arcu` only stores an `AtomicPtr<T>`, its `Sync`-ness is invariant to `T`: if `T: !Sync`, then `Arcu<T, P>` will implement `Sync`.

The following snippet is able to exploit this property to share a `RefCell` between threads and trigger UB (which Miri is able to detect):

```rust
struct Thing {
    rcu: Arcu<std::cell::RefCell<u8>, GlobalEpochCounterPool>,
}

impl Thing {
    fn send(&self) {
        std::thread::scope(|scope| {
            scope.spawn(|| {
                let mut ref_mut = self.rcu.read();
                let mut ref_mut = ref_mut.borrow_mut();
                *ref_mut = 1;
                println!("{}", ref_mut);
            });

            let mut ref_mut = self.rcu.read();
            let mut ref_mut = ref_mut.borrow_mut();
            *ref_mut = 2;
            println!("{}", ref_mut);
        });
    }
}

let thing = Thing {
    rcu: Arcu::new(std::cell::RefCell::new(0), GlobalEpochCounterPool)
};

thing.send();
```

It should be sufficient to add a `PhantomData<T>` within `Arcu` to make its `Sync`- and its `Send`-ness covariant.

**Comment by Skgland** [comment]
Thanks for the report.

The requirement needs to be even stronger as the `AtomicPtr<T>` is actually an `Arc<T>` so `Arcu` needs `PhantomData<Arc<T>>`.

**Comment by Skgland** [comment]
I have published 0.1.2 with a fix for this and also yanked version 0.1.1
